### ISSUE TITLE
Number 6 character

### ISSUE THREAD

**Issue by solarmon** [coopzone-dc/GotekLEDC68#2]
Hi.

Happy New Year!

This is probably/possibly an issue with the TM16xx library, but is there a way to change the 6 digit to look more like a number six?

At the moment, when using the displayNum() function, the top segment is not highlighted for a number 6, so it looks more like a lowercase 'b' instead.

Thank you.

**Comment by coopzone-dc** [comment]
I see what you mean. you could add a top bar to the 6, that may be suitable for you. But it's very limited as to what you can do with just 7 segments !

The display is broken down into a bit mask:
bit.   value on       segment
          in hex

0.      1h                top segment of LED
1.       2h               right-top segment
2.      4h               right-bottom segment
3.      8h               bottom of LED
4.      10h             bottom-left 
5.       20h           top-left
6.       40h           middle bar of LED
7.        80h          in theory this would be the DP, but is not present on many displays.

If you look at round line 35 of the TM1651.ccp file you will see an array called NumTab:
static int8_t NumTab[] = 
{ 0x3f, 0x06, 0x5b, 0x4f, 0x66, 0x6d, 0x7c, 0x07, 0x7f, 0x6f,
  0x77, 0x7c, 0x39, 0x5e, 0x79, 0x71,
  0x00, 0x63, 0x5c, 0x01, 0x40, 0x08
}; //numbers 0-9, A-F, special chars

The value of 0x7c in the first row is the number 6 displayed on led (if you break it down to binary:
80h 40h 20h 10h 8h 4h 2h 1h
0      1      1      1     1    1    0   0
so 40h+20h+10h+8h+4h  = 7ch
if you want to "add" in the top bar of the led to try it, your would have:
80h 40h 20h 10h 8h 4h 2h 1h
0      1      1      1     1    1    0   1
so 40h+20h+10h+8h+4h+1h = 7dh

just swap the 0x7c to be 0x7d, the new line would be:

static int8_t NumTab[] = 
{ 0x3f, 0x06, 0x5b, 0x4f, 0x66, 0x6d, **0x7d**, 0x07, 0x7f, 0x6f,
  0x77, 0x7c, 0x39, 0x5e, 0x79, 0x71,
  0x00, 0x63, 0x5c, 0x01, 0x40, 0x08
}; //numbers 0-9, A-F, special chars

I think i would leave it to choice if people wanted a 6 with the top bar or not.

Derek

**Comment by solarmon** [comment]
Hi,

Thanks for the comprehensive response and explanation. 

As per the wiki page at:

https://en.wikipedia.org/wiki/Seven-segment_display

a number six is listed as being 0x7D, as you have just calculated. And a 'b' is 0x7C, which is what is also currently for a number 6 in TM1651.cpp. So it cannot be that two different characters have the same value and look the same.

EDIT: Although I have just noticed it also states:

"The numerals 6 and 9 may also be represented without a 'tail'"

I've made the changes to TM1651.cpp but it still doesn't work for me for some reason. It probably has to do with my Arduino environment and it caching the compiled libraries - so I need to figure that out.

EDIT: I figured it out - I was editing the wrong TM1651.cpp file. It is now working as expected. Thank you!
